cdap-app-fabric/src/main/java/io/cdap/cdap/internal/operations/OperationLifecycleService.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.operations;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.internal.app.store.OperationRunDetail;
+import io.cdap.cdap.spi.data.StructuredTableContext;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+/**
+ * Service that manages lifecycle of Operation.
+ */
+public class OperationLifecycleService {
+
+  private final TransactionRunner transactionRunner;
+
+  @Inject
+  public OperationLifecycleService(TransactionRunner transactionRunner) {
+    this.transactionRunner = transactionRunner;
+  }
+
+  //TODO(samik) add unit test
+  /**
+   * Scan operations in a namespace.
+   *
+   * @param request scan request including filters and limit
+   * @param txBatchSize batch size of transaction
+   * @param consumer {@link Consumer} to process each scanned run
```

Comment:
javadoc should have `@return` as well

Reply:
though returning whether the limit was reached is kind of a weird thing to return, not sure how it would be used. I would either make it void or return the total number of operations scanned.

Reply:
The boolean would be used in the handler to find if all the operations are scanned and send last operation id accordingly. We can use the total number of operation scanned also but wanted to keep this analogous to the application scan

see: https://github.com/cdapio/cdap/blob/550a8f3940500ddeabb52330f4958366966acc50/cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java#L307

Reply:
hm I see, that's ok then